heaps/heap_sort.py

```diff
@@ -1,8 +1,28 @@
 
+def heapify(arr, n, i):
+    largest = i
+    l = 2 * i + 1
+    r = 2 * i + 2
+    if l < n and arr[l] > arr[largest]:
+        largest = l
+    if r < n and arr[r] > arr[largest]:
+        largest = r
+    if largest != i:
+        arr[i], arr[largest] = arr[largest], arr[i]
+        heapify(arr, n, largest)
 
-def heap_sort(list):
+def heap_sort(arr):
     """ This method uses a heap to sort an array.
-        Time Complexity:  ?
-        Space Complexity: ?
+        Time Complexity:  O(n)
+        Space Complexity: O(n)
```

Comment:
👀 Since each call to your `heapify` helper takes O(log n) time, and we are calling it 2n times here, the time complexity of `heap_sort` will be O(2n log n) → O(n log n). Note that for a general sorting algorithm (like heap sort), O(n log n) is the theoretical _best_ we can do!

For space complexity, you avoided needing to allocate the internal store that `MinHeap` used, but since `heapify` is implemented recursively, the stack will grow with the number of layers in the heap. So each `heapify` call will have a stack with O(log n) levels. Each is used only for the duration of a single `heapify` call, so they aren't cumulative, resulting in an overall space complexity of O(log n). But if you modified your `heapify` to be iterative, the space complexity would be constant O(1)!